org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -2046,182 +2046,25 @@ private static boolean getSplitVariableProposals(IInvocationContext context, AST
-					commentToken= buffer.getText(extendedStatementStart, oldVarDecl.getStartPosition() - extendedStatementStart);
-				}
-				codeDeclaration= commentToken + codeDeclaration;
-				nType= (Type) rewrite.createStringPlaceholder(codeDeclaration.trim(), type.getNodeType());
-			} else {
-				int extendedStartPositionDeclaration= cup.getExtendedStartPosition(oldVarDecl);
-				int firstFragmentStart= ((ASTNode) oldVarDecl.fragments().get(0)).getStartPosition();
-				String codeDeclaration= buffer.getText(extendedStartPositionDeclaration, firstFragmentStart - extendedStartPositionDeclaration);
-				nType= (Type) rewrite.createStringPlaceholder(codeDeclaration.trim(), type.getNodeType());
-			}
-
-			VariableDeclarationFragment newFrag= ast.newVariableDeclarationFragment();
-			VariableDeclarationStatement newVarDec= ast.newVariableDeclarationStatement(newFrag);
-			newVarDec.setType(nType);
-			newFrag.setName(ast.newSimpleName(oldFragments.get(0).getName().getIdentifier()));
-			newFrag.extraDimensions().addAll(DimensionRewrite.copyDimensions(oldFragments.get(0).extraDimensions(), rewrite));
-
-			for (int i= 1; i < oldFragments.size(); i++) {
-				VariableDeclarationFragment oldFragment= oldFragments.get(i);
-				newFrag= ast.newVariableDeclarationFragment();
-				newFrag.setName(ast.newSimpleName(oldFragment.getName().getIdentifier()));
-				newFrag.extraDimensions().addAll(DimensionRewrite.copyDimensions(oldFragment.extraDimensions(), rewrite));
-				newVarDec.fragments().add(newFrag);
-			}
-			newVarDec.modifiers().addAll(ASTNodeFactory.newModifiers(ast, oldVarDecl.getModifiers()));
-
-			ListRewrite listRewriter= rewrite.getListRewrite(statementParent, (ChildListPropertyDescriptor) property);
-			listRewriter.insertBefore(newVarDec, statement, null);
-
-			rewrite.remove(oldVarDecl, null);
-
-			resultingCollections.add(proposal);
-			return true;
-		}
-
-		int insertIndex= list.indexOf(statement);
-		ITypeBinding binding= fragment.getInitializer().resolveTypeBinding();
-		Expression placeholder= (Expression) rewrite.createMoveTarget(fragment.getInitializer());
-		if (placeholder instanceof ArrayInitializer && binding != null && binding.isArray()) {
-			ArrayCreation creation= ast.newArrayCreation();
-			creation.setInitializer((ArrayInitializer) placeholder);
-			final ITypeBinding componentType= binding.getElementType();
-			Type type= null;
-			if (componentType.isPrimitive()) {
-				type= ast.newPrimitiveType(PrimitiveType.toCode(componentType.getName()));
-			} else {
-				type= ast.newSimpleType(ast.newSimpleName(componentType.getName()));
+			VariableDeclarationFixCore fix= VariableDeclarationFixCore.createSplitVariableFix(context.getASTRoot(), node);
+			if (fix != null) {
+				Image image= JavaPluginImages.get(JavaPluginImages.IMG_CORRECTION_LOCAL);
+				FixCorrectionProposal proposal= new FixCorrectionProposal(fix, null, IProposalRelevance.CONVERT_FOR_LOOP_TO_ENHANCED, image, context);
```

Comment:
I don't feel this change is of the same relevance as a convert to enhanced for loop quick assist.  I might suggest you add a new constant for SPLIT_JOIN_VARIABLE_DECLARATION and make it so CONVERT_FOR_LOOP_TO_ENHANCED is higher priority.  This will make it clearer to anyone looking at priorities instead of having to know a relevance has been reused.